### PR TITLE
dev-libs/qtkeychain: add Qt6 support

### PR DESCRIPTION
**Pull request by CyberTailor**

### CODE REVIEW

**Comment by CyberTailor** [comment]
Explicit dependency on `dev-qt/qtkeychain[qt5]` for revdeps is not needed right now because `qt6` USE flag is still masked.

File: dev-libs/qtkeychain/qtkeychain-0.13.2.ebuild [hunk 1]
```diff
@@ -1,47 +1,85 @@
...
 else
 	inherit git-r3
-	EGIT_REPO_URI="${HOMEPAGE}.git"
+	EGIT_REPO_URI="https://github.com/frankosterfeld/${PN}.git"
 fi
 
 LICENSE="BSD-2"
 SLOT="0/1"
-IUSE="gnome-keyring"
+IUSE="gnome-keyring +qt5 qt6"
```
Comment (a17r) [inline dev-libs/qtkeychain/qtkeychain-0.13.2.ebuild:21]: please rename to `keyring`

File: dev-libs/qtkeychain/qtkeychain-0.13.2.ebuild [hunk 2]
```diff
@@ -1,47 +1,85 @@
...
 DEPEND="
-	dev-qt/qtcore:5
-	dev-qt/qtdbus:5
+	qt5? (
+		dev-qt/qtcore:5
+		dev-qt/qtdbus:5
+	)
+	qt6? (
+		dev-qt/qtbase:6[dbus]
+	)
```
Comment (a17r) [inline dev-libs/qtkeychain/qtkeychain-0.13.2.ebuild:34]: single deps on single line please

File: dev-libs/qtkeychain/qtkeychain-0.13.2.ebuild [hunk 3]
```diff
@@ -1,47 +1,85 @@
...
+}
+
+qt_get_mkspecsdir() {
+	echo "${EPREFIX}"/usr/$(get_libdir)/${MULTIBUILD_VARIANT}/mkspecs
+}
+
+my_src_configure() {
 	local mycmakeargs=(
-		-DBUILD_WITH_QT6=OFF
+		-DECM_MKSPECS_INSTALL_DIR=$(qt_get_mkspecsdir)
```
Comment (a17r) [inline dev-libs/qtkeychain/qtkeychain-0.13.2.ebuild:65]: why do we need a function if only used once?

File: dev-libs/qtkeychain/qtkeychain-0.13.2.ebuild [hunk 4]
```diff
@@ -1,47 +1,85 @@
...
+		-DBUILD_WITH_QT6=$(with_qt6)
```
Comment (a17r) [inline dev-libs/qtkeychain/qtkeychain-0.13.2.ebuild:66]:
```suggestion
		if [[ ${MULTIBUILD_VARIANT} == qt6 ]]; then
			mycmakeargs+=( -DBUILD_WITH_QT6=ON )
		else
			mycmakeargs+=( -DBUILD_WITH_QT6=OFF )
		fi
```

File: dev-libs/qtkeychain/qtkeychain-0.13.2.ebuild [hunk 5]
```diff
@@ -1,47 +1,85 @@
...
+		qt5) echo OFF ;;
+		qt6) echo ON ;;
+	esac
+}
+
+qt_get_mkspecsdir() {
+	echo "${EPREFIX}"/usr/$(get_libdir)/${MULTIBUILD_VARIANT}/mkspecs
+}
+
+my_src_configure() {
```
Comment (a17r) [inline dev-libs/qtkeychain/qtkeychain-0.13.2.ebuild:63]: we usually make these as nested function within `src_configure`

**Review by a17r — CHANGES_REQUESTED** [review]

**Comment by a17r** [comment]
Ah, also if you rebase on top of current `HEAD` the CI error about `qt6` flag will go away. :)

File: dev-libs/qtkeychain/qtkeychain-0.13.2.ebuild (hunk 1 above)
Comment (CyberTailor) [reply dev-libs/qtkeychain/qtkeychain-0.13.2.ebuild]: fixed

File: dev-libs/qtkeychain/qtkeychain-0.13.2.ebuild (hunk 2 above)
Comment (CyberTailor) [reply dev-libs/qtkeychain/qtkeychain-0.13.2.ebuild]: fixed

File: dev-libs/qtkeychain/qtkeychain-0.13.2.ebuild (hunk 4 above)
Comment (CyberTailor) [reply dev-libs/qtkeychain/qtkeychain-0.13.2.ebuild]: applied

File: dev-libs/qtkeychain/qtkeychain-0.13.2.ebuild (hunk 5 above)
Comment (CyberTailor) [reply dev-libs/qtkeychain/qtkeychain-0.13.2.ebuild]: fixed

**Comment by CyberTailor** [comment]
> The only thing left doing now is revbumping 0.14.0 for the RDEPEND changes.

Done.

**Comment by a17r** [comment]
Thanks!

**Pull request closed without merging**